### PR TITLE
Citizenship by grant

### PR DESCRIPTION
**Pull request by Br3nda**
Closes https://github.com/ServiceInnovationLab/PresenceChecker/issues/6

* Added Citizen-by-grant presence in NZ requirements algorithm
* Change `age` to be changing by `DAY`
* Uses `day` branch from Open Fisca
* New Variables
  - `citizenship__citizenship_by_grant_may_be_authorized`
  - `citizenship__meets_minimum_presence_requirements`
  - `days_present_in_new_zealand_in_preceeding_5_years`
  - `days_present_in_new_zealand_in_preceeding_year`
  - `present_in_new_zealand`
  - `is_of_full_capacity`
  - `immigration__holds_indefinite_stay_visa`
  - `citizenship__is_of_good_character`
  - `citizenship__has_sufficient_knowledge_of_the_responsibilities_and_privileges_attaching_to_nz_citizenship`
  - `citizenship__has_sufficient_knowledge_of_the_english_language`
  - `citizenship__intends_to_reside_in_nz`
  - `citizenship__intends_nz_employment`
 - `citizenship__intends_international_service`
 - `citizenship__intends_crown_service`

- - - -

These changes
- Add a new feature: ability to calculate days when an applicant meets the minimum presence in NZ requirements for citizenship by grant

### CODE REVIEW

**Comment by Br3nda** [comment]
Okay, i've got this PR deployed to a new heroku app

and i set up a test edition of Rapu Ture - that lets you browse the rules : https://rulestestnz.herokuapp.com/variables/citizenship__citizenship_by_grant_may_be_authorized

File: openfisca_aotearoa/variables/acts/citizenship/citizenship.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    label = u"Applicant was present in New Zealand for a min of 1,350 days during the 5 years immediately preceding the date of application"
+    reference = "http://www.legislation.govt.nz/act/public/1977/0061/latest/DLM443855.html"
+
+    def formula(persons, period, parameters):
+        # persons('immigration__entitled_to_stay_indefinitely', period) * \
+        # (ii) for at least 240 days in each of those 5 years,—
+        # being days during which the applicant was entitled in terms of the Immigration Act 2009 to be in New Zealand indefinitely
+        # for p in [period.offset(offset) for offset in range(-365, 1)]:
+        return persons('citizenship__meets_5_year_presence_requirement', period) * \
+            persons('citizenship_meets_each_year_minimum_presence_requirements', period)
```
Comment (dlouise64) [inline openfisca_aotearoa/variables/acts/citizenship/citizenship.py]: @Br3nda I think you intended to have `citizenship__meets_each_year_minimum_presence_requirements`. Missing extra underscore

Comment (Br3nda) [reply openfisca_aotearoa/variables/acts/citizenship/citizenship.py]: fixed

**Comment by Br3nda** [comment]
Excellent.. I'll merge tomorrow morning.

**Pull request merged**